contracts/telospooldex/telospooldex.hpp

```diff
@@ -0,0 +1,131 @@
+#pragma once
+
+#include <vapaee/base/base.hpp>
+#include <vapaee/dex/tables.hpp>
+#include <vapaee/pool/tables.hpp>
+
+#define ARITHMETIC_PRECISION 8
+
+#define ERR_MARKET_NOT_FOUND    "market not found"
+#define ERR_POOL_NOT_FOUND      "pool not found"
+#define ERR_POOL_EXISTS         "pool already exists"
+#define ERR_TOKEN_NOT_REG       "token not registered"
+#define ERR_COMM_SYM_NOT_MATCH  "commodity symbols don\'t match"
+#define ERR_CURR_SYM_NOT_MATCH  "currency symbols don\'t match"
+#define ERR_ACCOUNT_NOT_FOUND   "account not found"
+#define ERR_MEMO_PARSING        "incorrect memo format"
+#define ERR_ATTEMPT_NOT_FOUND   "fund attempt not found"
+#define ERR_NOT_FUNDER          "not funder"
+#define ERR_EMPTY_PATH          "path is empty"
+#define ERR_INCORRECT_CONVERTER "this is not the converter you need"
+#define ERR_POOL_NOT_FUNDED     "unfunded pool"
+#define ERR_RECIPIENT_NOT_FOUND "recipient not found"
+#define ERR_BAD_DEAL            "total less than minimun"
+#define ERR_CONVERTER_NOT_FOUND "can't find converter"
+
+#define PROTO_VERSION "openpool.v1"_n
+
+#define ECHO_CONTRACT "echocontract"_n
```

Comment:
I already bought the name echocontract:
https://telos.bloks.io/account/echocontract